src/main/java/org/teamresistance/frc/subsystem/drive/DriveToXController.java

```diff
@@ -0,0 +1,59 @@
+package org.teamresistance.frc.subsystem.drive;
+
+import org.strongback.control.SoftwarePIDController;
+import org.teamresistance.frc.Feedback;
+import org.teamresistance.frc.subsystem.Controller;
+import org.teamresistance.frc.util.SynchronousPID;
+
+/**
+ * Created by shrey on 1/25/2017.
+ */
+public class DriveToXController implements Controller<Drive.Signal> {
+  private final SynchronousPID rotationPID;
+  private static final double ROTATION_TOLERANCE = 1;
+  private static final double ROTATION_KP = 0.008;
+
+  private double xDistPingSensor;
+  private final double hackTargetX;
+
+  private final SynchronousPID xTranslationPID;
+  private static final double X_TRANSLATION_TOLERANCE = 10;
+  private static final double X_TRANSLATION_KP = 0.02;
+
+  private static final double ADDED_POWER = 0.2;
+
+  public DriveToXController(double targetX) {
+    hackTargetX = targetX;
+
+    this.rotationPID = new SynchronousPID(SoftwarePIDController.SourceType.DISTANCE,
```

Comment:
Rather than using a brand new rotation PID, you can hold a reference to the `DriveHoldingAngleController` and use *that* to compute your rotation output. Saves you from having to keep track of the gains in multiple places. See the `DriveStrafingController` for reference.